src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/csip/CSIPService.kt

```diff
@@ -149,48 +152,78 @@ class CSIPService(
+  private fun CSIPReport.addNonMandatoryFields(currentRequest: UpsertCSIPRequest) {
+    // TODO Add release date -> Confirmed Release Date (from External Movements)
+    // releaseDate = latestBooking.releaseDetail.releaseDate,
+    currentRequest.reportDetailRequest?.let { reportDetailRequest ->
+      involvement = reportDetailRequest.involvementCode?.let { lookupInvolvement(it) }
+      concernDescription = reportDetailRequest.concern
+      knownReasons = reportDetailRequest.knownReasons
+      otherInformation = reportDetailRequest.otherInformation
+      saferCustodyTeamInformed = reportDetailRequest.saferCustodyTeamInformed
+      referralComplete = reportDetailRequest.referralComplete
+      referralCompletedBy = reportDetailRequest.referralCompletedBy
+      referralCompletedDate = reportDetailRequest.referralCompletedDate
+      if (reportDetailRequest.factors.isNotEmpty()) {
+        addOrInsertFactor(request = currentRequest)
       }
-      decision?.let { setDecisionData(decision) }
+    }
+
+    currentRequest.saferCustodyScreening?.let { saferCustodyScreening ->
+      outcome = lookupOutcome(saferCustodyScreening.scsOutcomeCode)
+      reasonForDecision = saferCustodyScreening.reasonForDecision
+      outcomeCreateUsername = saferCustodyScreening.recordedBy
+      outcomeCreateDate = saferCustodyScreening.recordedDate
+    }
 
-      caseManager = caseManager
-      reasonForPlan = planReason
-      firstCaseReviewDate = firstCaseReviewDate
-      // plans = mutableListOf()
-      // reviews = mutableListOf()
+    currentRequest.investigation?.let { investigation ->
+      staffInvolved = investigation.staffInvolved
+      evidenceSecured = investigation.evidenceSecured
+      reasonOccurred = investigation.reasonOccurred
+      usualBehaviour = investigation.usualBehaviour
+      trigger = investigation.trigger
+      protectiveFactors = investigation.protectiveFactors
+      // TODO interviews = mutableListOf()
+    }
+    currentRequest.decision?.let { setDecisionData(currentRequest.decision) }
+
+    caseManager = currentRequest.caseManager
+    reasonForPlan = currentRequest.planReason
+    firstCaseReviewDate = currentRequest.firstCaseReviewDate
+    // TODO plans = mutableListOf()
+    // TODO reviews = mutableListOf() and attendees
+
+    lastModifiedUsername = currentRequest.auditDetails.modifyUsername
+    lastModifiedDateTime = currentRequest.auditDetails.modifyDatetime
+  }
 
-      lastModifiedUsername = auditDetails.modifyUserId
-      lastModifiedDateTime = auditDetails.modifyDatetime
+  private fun CSIPReport.addOrInsertFactor(request: UpsertCSIPRequest) {
+    request.reportDetailRequest?.factors?.forEach { factorRequest ->
+      factorRequest.id?.let {
+        factors.find { it.id == factorRequest.id }
+          ?.also {
+            it.type = lookupFactorType(factorRequest.typeCode)
+            it.comment = factorRequest.comment
+            it.createUsername = factorRequest.auditDetails.createUsername
```

Comment:
Update will not have update createUserName - I will change this and add tests for the update in the next PR